buf/registry/module/v1beta1/branch_service.proto

```diff
@@ -98,3 +117,117 @@ message ListBranchesResponse {
   // The listed Branches.
   repeated Branch branches = 2;
 }
+
+message PushCommitsRequest {
+  // A request to push a Commit to a Branch.
+  message Value {
+    // The Branch to push the Commit to.
+    BranchRef branch_ref = 1 [(buf.validate.field).required = true];
+    // The ID of the Commit to push to the Branch.
+    //
+    // This Commit will become the latest Commit on the Branch.
+    //
+    // If the Commit already exists on the Branch, an error is returned.
+    string commit_id = 2 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.uuid = true
+    ];
+  }
+  // The requests to push a Commit to a Branch.
+  //
+  // Commits will be pushed to Branches in the order provided.
+  repeated Value values = 1 [(buf.validate.field).repeated.min_items = 1];
+}
+
+message PushCommitsResponse {
+  message Value {
+    // The Branch to which the Commit was pushed.
+    Branch branch = 1 [(buf.validate.field).required = true];
+    // The Commit that was pushed to the Branch.
+    Commit commit = 2 [(buf.validate.field).required = true];
+  }
+  // The Branches that were pushed to, in the order of the request.
+  repeated Value values = 1 [(buf.validate.field).repeated.min_items = 1];
+}
+
+message PatchBranchHistoriesRequest {
+  // A request to create a new Branch with the expected history.
+  //
+  // At the end of this request, the Branch's history looks like: [...imported, ...added].
+  message CreateBranchHistory {
+    // The module for which to create the Branch.
+    ModuleRef module_ref = 1 [(buf.validate.field).required = true];
+    // The name of the branch to create.
+    string branch_name = 2 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.max_len = 250
+    ];
+    // The ID of another Commit on another Branch from which to import Branch history. The history
+    // of any Branch that contains this Commit will be imported into the newly created Branch, up to
+    // and including this Commit.
+    //
+    // This is primarily designed for use in forking a Branch from an existing Branch instead of
+    // providing the new Branch's entire history.
+    optional string import_from_commit_id = 3 [(buf.validate.field).string.uuid = true];
```

Comment:
Why does this have field presence?

Reply:
Would we prefer to just treat empty value as not set instead? I.e., only use field presence when the default value isn't sufficient?